src/server/server.ts

```diff
@@ -528,12 +518,72 @@ namespace ts.server {
         }
     }
 
+    export class DefaultEventSender implements EventSender {
+        constructor(protected host: ServerHost,
+            protected byteLength: (buf: string, encoding?: string) => number,
+            protected logger: Logger,
+            protected canUseEvents: boolean) { }
+
+        public event = <T>(body: T, eventName: string) => {
+            const ev: protocol.Event = {

```

Comment:
It is a bit awkward that this sender duplicates the implementation of `Session.prototype.event`.

If someone subclasses `Session` and overwrites `send`, the `event` implementation in `Session` will call the overridden `send`. This "default" sender will not.

Overall this feels like an imperfect way of factoring out the default behavior. Any suggestions for how to do this better?